packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.ts

```diff
@@ -418,7 +421,7 @@ async function getS3ResourceNameFromMeta(amplifyMeta: $TSAny): Promise<string |
  * @returns { resourceName => resourceData in Amplify metafile}
  */
 function getS3ResourcesFromAmplifyMeta(amplifyMeta: $TSMeta): Record<string, $TSAny> | undefined {
-  if (!amplifyMeta.hasOwnProperty(AmplifyCategories.STORAGE)) {
+  if (!Object.prototype.hasOwnProperty.call(amplifyMeta, AmplifyCategories.STORAGE)) {
```

Comment:
Can this pattern be simplified? Something like...

```suggestion
  if (!amplifyMeta?.[AmplifyCategories.STORAGE]) {
```

Reply:
Those two lines don't have quite the same meaning, but I would guess that that approach would work in practice in this codebase. I think I would lean towards preserving the `Object.prototype` route for this commit, and then address a refactor in a follow on?